src/v/config/property.h

```diff
@@ -123,6 +132,152 @@ class property : public base_property {
     };
 };
 
+template<typename T>
+concept has_type_name = requires(T x) {
+    x.type_name();
+};
+
+template<typename T>
+concept is_collection = requires(T x) {
+    typename T::value_type;
+    {x.size()};
+};
+
+template<typename T>
+consteval std::string_view property_type_name_inner() {
```

Comment:
Can the implementation details go into a `namespace detail {}`.

Reply:
Done